File: src/main/java/com/anatartari/tatamecontrolapi/infra/database/entity/RegistrationEntity.java
package com.anatartari.tatamecontrolapi.infra.database.entity;

import com.anatartari.tatamecontrolapi.core.model.StatusRegistrationEnum;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Entity
@Table(name = "registration")
public class RegistrationEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "student_id", nullable = false)
    private StudentEntity student;

    @ManyToOne
    @JoinColumn(name = "sport_id", nullable = false)
    private SportEntity sport;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private StatusRegistrationEnum status;

    @ManyToMany
    @JoinTable(name = "registration_payment",
            joinColumns = @JoinColumn(name = "registration_id"),
            inverseJoinColumns = @JoinColumn(name = "payment_id"))
    private List<PaymentEntity> payments = new ArrayList<>();

}

File: src/registration/domain/entities/registration.entity.ts
import { Entity, ManyToOne, Column, Unique, JoinTable, ManyToMany, JoinColumn } from 'typeorm';
import { Student } from '../../../student/domain/entities/student.entity';
import { Sport } from '../../../sport/domain/entities/sport.entity';
import { Payment } from '../../../payment/domain/entities/payment.entity';
import { BaseEntity } from '../../../global/entities/base.entity';
import { RegistrationStatusEnum } from '../../../global/enums/registrationStatus.enum';

@Entity()
@Unique(['student', 'sport'])
export class Registration extends BaseEntity {

    @ManyToOne(() => Student, { onDelete: 'CASCADE' })
    @JoinColumn({ name: 'studentId' })
    student: Student;

    @ManyToOne(() => Sport, { onDelete: 'CASCADE' })
    @JoinColumn({ name: 'sportId' })
    sport: Sport;

    @Column({ 
        type: 'varchar', 
        length: 50,
        enum: RegistrationStatusEnum,
        default: RegistrationStatusEnum.ACTIVE
    })
    status: RegistrationStatusEnum;

    @ManyToMany(() => Payment, (payment) => payment.registrations)
    @JoinTable({
        name: 'registrationPayment',
        joinColumn: {
            name: 'registrationId',
            referencedColumnName: 'id'
        },
        inverseJoinColumn: {
            name: 'paymentId',
            referencedColumnName: 'id'
        }
    })
    payments: Payment[]
}

